students/eowyn/session12/Mailroom/mailroom/test/test_auth_user.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+
+import pytest
+from unittest import mock
+from mailroom.authorized_users import UserLogger
+
+
+@pytest.fixture
+def decorated_function():
+    ul = UserLogger()
+
+    @ul.record_user
+    def some_funct():
+        pass
+    return some_funct, ul
+
+
+@mock.patch('builtins.input')
+class TestAuthUser():
+
+
+
+    def test_auth_user_decorator(self, mocked_input, decorated_function):
+        decorated_function[0]()
+        mocked_input.assert_called_once()
+
+    def test_auth_user_called_with(self, mocked_input, decorated_function):
+        decorated_function[0]()
+        mocked_input.assert_called_once_with("Please enter user name:\n")
+
```

Comment:
You don't need both of these two tests. The second one is more complete, so I would just use that one.